packages/cli/test/integration/publicApi/executions.test.ts

```diff
@@ -330,54 +304,67 @@ describe('GET /executions', () => {
 		} = response.body.data[0];
 
 		expect(id).toBeDefined();
-		expect(finished).toBe(false);
-		expect(mode).toEqual(errorExecution.mode);
+		expect(finished).toBe(filterStatus == "success");
+		expect(mode).toEqual(executions.get(filterStatus).mode);
 		expect(retrySuccessId).toBeNull();
 		expect(retryOf).toBeNull();
 		expect(startedAt).not.toBeNull();
 		expect(stoppedAt).not.toBeNull();
-		expect(workflowId).toBe(errorExecution.workflowId);
-		expect(waitTill).toBeNull();
+		expect(workflowId).toBe(executions.get(filterStatus).workflowId);
+		if (filterStatus == "waiting") {
+			expect(new Date(waitTill).getTime()).toBeGreaterThan(Date.now() - 1000);	
+		} 
+		else {
+			expect(waitTill).toBeNull();
+		}
```

Comment:
same here for this kind of if else in test

Reply:
How would I go about testing something like this?

Reply:
I must admit it will be tricky to make it cleane :)